tests/v2/test_waku_peer_exchange.nim

```diff
@@ -126,14 +126,17 @@ procSuite "Waku Peer Exchange":
     await allFutures([node1.start(), node2.start(), node3.start()])
     await allFutures([node1.startDiscv5(), node2.startDiscv5()])
 
+    # node2 can be connected, so will be returned peer exchange
+    require (await node1.peerManager.connectRelay(node2.switch.peerInfo.toRemotePeerInfo()))
+
+    # Give disv5 some time to discover each other
+    await sleepAsync(5000.millis)
```

Comment:
Is there a way to limit these "arbitrary" sleeps in our tests and rather check for the state (e.g. in a loop with some smaller sleep duration) or would that be an overkill? IMO that could potentially speed up some test runs and at the same time make them more resilient in case the node is slower.

Reply:
indeed, totally agree that in some tests we should follow this pattern, waitFor instead of arbitrary sleep. we would still need a timeout but ofc tests would finish before, and we can cut some time.

perhaps not for this specific test, but will take it into account for further tests/refactors thanks!